core/common/src/main/java/alluxio/heartbeat/HeartbeatContext.java

```diff
@@ -125,7 +125,7 @@ public static synchronized Class<? extends HeartbeatTimer> getTimerClass(String
    * @param timerClass the timer class to use for the executor thread
    */
   @SuppressWarnings("unused")
-  private static synchronized void setTimerClass(String name,
+  protected static synchronized void setTimerClass(String name,
```

Comment:
This class is `final`, so protected access does not make sense. Package-private access will do here. Also, please add a `@VisibleForTesting` annotation.

Reply:
done~